jetbrains-core/build.gradle

```diff
@@ -44,7 +44,7 @@ sourceSets {
 }
 
 test {
-    systemProperty("log.dir", "${org.jetbrains.intellij.Utils.systemDir(project.intellij.sandboxDirectory, true)}/logs")
+    systemProperty("log.dir", "${project.intellij.sandboxDirectory}-test/logs")
```

Comment:
surprised they didn't opt to create a new sandbox specifically for ui tests

Reply:
I checked, they did:

ls jetbrains-ultimate/build/idea-sandbox/
config         config-test    config-uiTest  plugins        plugins-test   plugins-uiTest system

but this is actually for unit tests only.

Non-headless IDEs log correctly. This hack is for headless only to force logging to turn on.